Library/Homebrew/download_strategy.rb

```diff
@@ -262,6 +262,16 @@ def initialize(name, resource)
   end
 
   def fetch
+    insecure_redirects = @url.include?("https://downloads.sourceforge.net") || @url.include?("https://download.gnome") || @url.include?("https://www.apache.org/dyn/")
+    if insecure_redirects && ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
+      puts <<-EOS.undent
+        This URL may redirect to http and the download may not be protected by SSL/TLS.
+
+        You can disable insecure redirect by setting the "HOMEBREW_NO_INSECURE_REDIRECT" environmental variable.
```

Comment:
Wasn't sure either way on this. The latter is used in a few places:

https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/cmd/doctor.rb#L793

https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/External-Commands.md#shell-scripts